app/models/coin.rb

```diff
@@ -211,6 +226,84 @@ def sparkline
     end
   end
 
+  def token_metrics_data(metric_type)
+    return nil unless has_token_metrics?
+
+    url = "#{ENV.fetch('COINFI_POSTGREST_URL')}/metrics_chart_view?coin_key=eq.#{coin_key}&metric_type=eq.#{metric_type}&select=percentage,date"
+    response = HTTParty.get(url)
+    JSON.parse(response.body)
```

Comment:
Does it make more sense to cache this method's return value rather than all of the individual accessor methods?

Basically the accessor methods (i.e. `exchange_supply_data`) just calls this master method and retrieves a hash key from it, so by consolidating the cache here it reduces complexity and also prevents multiple calls to this endpoint.